File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
    const stack = [];  // 스택으로 활용할 배열

    for (let char of s) {
        if (stack.length && stack[stack.length - 1] === char) {
            stack.pop();  // 짝이 맞으면 스택에서 제거
        } else {
            stack.push(char);  // 짝이 아니면 스택에 추가
        }
    }

    return stack.length ? 0 : 1;  // 스택에 남은 문자가 있으면 0, 없으면 1 반환
}

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    stackS = []
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.    
    
    
    for i in range(len(s)):
        if len(stackS) == 0:
            stackS.append(s[i])
        elif stackS[-1] == s[i]:
            stackS.pop()
        else:
            stackS.append(s[i])
    
    if len(stackS) == 0:
        return 1
    
    return answer